schema/agent/1.0.0/schema.json

```diff
@@ -0,0 +1,224 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "YAML Schema Validation for Azure Agent Configuration",
+    "description": "JSON Schema for validating the agent YAML configuration",
+    "type": "object",
+    "required": ["version", "name", "model", "instructions"],
+    "additionalProperties": false,
+    "properties": {
+      "version": {
+        "type": "string",
+        "description": "YAML schema version",
+        "const": "1.0.0"
+      },
+      "name": {
+        "type": "string",
+        "description": "Agent name",
+        "maxLength": 256
+      },
+      "description": {
+        "type": "string",
+        "description": "Agent description"
+      },
+      "metadata": {
+        "type": "object",
+        "description": "Metadata about the agent",
+        "additionalProperties": true,
+        "properties": {
+          "author": { 
+            "type": "string",
+            "description": "Author name"
+          },
+          "tag": { 
+            "type": "string",
+            "description": "Agent tag"
+          }
+        }
+      },
+      "id": {
+        "type": ["string", "null"],
+        "description": "Agent id is a unique identifier for the agent. Use this field to switch to a different agent. This filed will be auto generated after deploy."
```

Comment:
Suggested description: "Agent id is a unique identifier for the agent. This field is auto generated by Azure AI Foundry Agent Service after deployment."